lib/dal/src/cached_module.rs

```diff
@@ -288,271 +349,171 @@ impl CachedModule {
-    async fn insert_private_module(
+    async fn insert(
         ctx: &DalContext,
         module_details: &ModuleDetailsResponse,
         pkg_bytes: Arc<Vec<u8>>,
+        scoped_to_user_pk: Option<UserPk>,
     ) -> CachedModuleResult<Option<Self>> {
-        let bytes_clone = pkg_bytes.clone();
-        let pkg = slow_rt::spawn(async move { SiPkg::load_from_bytes(&bytes_clone) })?.await??;
-
-        let user_pk =
-            UserPk::from_raw_id(Ulid::from_string(module_details.owner_user_id.as_str())?);
-        dbg!(&user_pk);
-        let query = "
-            INSERT INTO cached_modules (
-                schema_id,
-                schema_name,
-                display_name,
-                category,
-                link,
-                color,
-                description,
-                component_type,
-                latest_hash,
-                created_at,
-                package_data,
-                scoped_to_user_pk
-            ) VALUES (
-                $1, $2, $3, $4, $5, $6,
-                $7, $8, $9, $10, $11, $12
-            ) RETURNING
-              id,
-              schema_id,
-              schema_name,
-              display_name,
-              category,
-              link,
-              color,
-              description,
-              component_type,
-              latest_hash,
-              created_at,
-              NULL::bytea AS package_data,
-              NULL::ident AS scoped_to_user_pk
-        ";
+        let query = format!(
+            "
+                INSERT INTO cached_modules (
+                    schema_id,
+                    schema_name,
+                    display_name,
+                    category,
+                    link,
+                    color,
+                    description,
+                    component_type,
+                    package_summary,
+                    latest_hash,
+                    created_at,
+                    package_data,
+                    scoped_to_user_pk
+                ) VALUES (
+                    $1, $2, $3, $4, $5, $6,
+                    $7, $8, $9, $10, $11, $12, $13
+                ) RETURNING
+                    {CACHED_MODULE_LIST_FIELDS}
+            "
+        );
 
         let Some(schema_id) = module_details.schema_id() else {
             warn!("builtin module {} has no schema id", module_details.id);
             return Ok(None);
         };
         let schema_id: SchemaId = schema_id.into();
 
-        let Some(pkg_schema) = pkg.schemas()?.first().cloned() else {
-            warn!("builtin module {} has no schema", module_details.id);
-            return Ok(None);
-        };
-
-        let Some(pkg_variant) = pkg_schema.variants()?.first().cloned() else {
-            warn!(
-                "builtin module {} has a schema with no variant",
-                module_details.id
-            );
+        let Some(package) = PackageData::load(&module_details.id, &pkg_bytes).await? else {
```

Comment:
This encompasses the "load the package data in slow_rt," the "deserialize and get the name/color/description/etc." It now also includes "get the JSON PackageSummary (with the socket count).